substratevm/src/com.oracle.graal.pointsto.standalone/src/com/oracle/graal/pointsto/standalone/heap/StandaloneImageHeapScanner.java

```diff
@@ -54,4 +58,15 @@ protected Class<?> getClass(String className) {
             return null;
         }
     }
+
+    @Override
+    protected ValueSupplier<JavaConstant> readHostedFieldValue(AnalysisField field, JavaConstant receiver) {
+        ValueSupplier<JavaConstant> ret = super.readHostedFieldValue(field, receiver);
+        if (ret.get() == null) {
+            JavaConstant constant = UninitializedStaticFieldValueReader.readUninitializedStaticValue(field, value -> universe.getSnippetReflection().forObject(value));
+            return ValueSupplier.eagerValue(constant);
```

Comment:
I think this is fine in the end since the doc on [`ConstantReflectionProvider.readFieldValue`](https://github.com/openjdk/jdk/blob/f43bb9feaa03008bad9708a4d7ed850d2532e102/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/ConstantReflectionProvider.java#L35) states that `@return the value of this field or {@code null} if the value is not available (e.g., because the field holder is not yet initialized).`, but you should probably me more defensive since `UninitializedStaticFieldValueReader.readUninitializedStaticValue` assumes this is a static field.

Reply:
I added a check for static field to avoid the potential assertion failure.

Reply:
This got merged by https://github.com/oracle/graal/commit/5e100c36e3da1797c7df22b6bc40c564864a3ed7 without this check. Please include it in the next PR.

Reply:
I see. Should I close this PR?

Reply:
Yes, the important parts are already merged.